src/generators/python/presets/Pydantic.ts

```diff
@@ -101,7 +101,7 @@ def custom_serializer(self, handler):
 @model_validator(mode='before')
 @classmethod
 def unwrap_${dictionaryModel?.propertyName}(cls, data):
-  json_properties = list(data.keys())
+  json_properties = list(data.model_dump().keys())
```

Comment:
[Before model validators are passed the raw input which is often a dict[str, Any] but could also be an instance of the model itself (e.g. if UserModel.model_validate(UserModel.construct(...)) is called) ](https://docs.pydantic.dev/2.9/concepts/validators/#model-validators)

your code have to support dict(how it was before) and pydantic model

Reply:
I've replaced this with a function that I believe will work for both the model and a dict now. Can you check if this looks useable?

Reply:
I guess you need to change
`list_keys(data) `to `cls.list_keys(data)`

Reply:
Updated it to be `cls.list_keys(data)` now